app/vmalert/config/fs.go

```diff
@@ -31,18 +34,39 @@ var (
 	fsRegistry   = make(map[string]FS)
 )
 
-// readFromFS parses the given path list and inits FS for each item.
-// Once inited, readFromFS will try to read and return files from each FS.
-// readFromFS returns an error if at least one FS failed to init.
+// readFromFSOrHTTP reads path either from filesystem or from http if path starts with http or https.
+// when reading from filesystem, parses the given path list and inits FS for each item.
+// Once initialed, readFromFSOrHTTP will try to read and return files from each FS.
+// readFromFSOrHTTP returns an error if at least one FS failed to init.
 // The function can be called multiple times but each unique path
-// will be inited only once.
+// will be initialed only once.
 //
-// It is allowed to mix different FS types in path list.
-func readFromFS(paths []string) (map[string][]byte, error) {
-	var err error
+// It is allowed to mix different FS types and url in path list.
+func readFromFSOrHTTP(paths []string) (map[string][]byte, error) {
 	result := make(map[string][]byte)
 	for _, path := range paths {
+		if isHTTPURL(path) {
+			// reads remote file via http or https, if url is given
+			resp, err := http.Get(path)
+			if err != nil {
+				return nil, fmt.Errorf("cannot fetch %q: %w", path, err)
+			}
+			data, err := io.ReadAll(resp.Body)
+			_ = resp.Body.Close()
+			if resp.StatusCode != http.StatusOK {
+				if len(data) > 4*1024 {
```

Comment:
why limiting to 4KB?
Although, limiting the body in this way would 99% result into parsing error.

Reply:
this limitation only works when response code is non-200.
Because I saw [the fix in vmagent](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4034) and thought it's a harmless behavior in here.